lib/bolt/transport/ssh/connection.rb

```diff
@@ -148,10 +148,15 @@ def disconnect
 
         def handled_sudo(channel, data, stdin)
           if data.lines.include?(Sudoable.sudo_prompt)
+            # rubocop:disable Style/GlobalVars
             if target.options['sudo-password']
               channel.send_data("#{target.options['sudo-password']}\n")
               channel.wait
               return true
+            elsif target.options['password'] && $future
```

Comment:
It might make sense to set this in the initializer, like `@sudo_password = target.options['sudo-password'] || target.options['password']` (gating that second part on `$future`). Then here we can just check in one place for whether we have `@sudo_password`.

Reply:
Updated.